[2025-10-05 04:24:24,989][root][INFO] - Configuration: experiment:
  data_path: ../data
  checkpoint_dir: ../checkpoints
  models:
  - 'inf'
  row_fraction: 1.0
  num_targets: 424
  filter_features: false
  optuna:
    direction: minimize
    n_trials: 2
    n_jobs: 2
model:
  model_type: default
  model_class: default
  trainer_class: default
  params:
    input_size: 1672
    output_size: 424
    input_len: 64
    output_len: 5
    batch_size: 16
    hidden_size: 128
    num_layers: 2
    dropout: 0.1
    epochs: 10
    patience: 5
    channels: 64
    blocks: 3
    kernel_size: 5
    num_conv_layers: 2
    tcn_kernel: 3
    decom_kernel_size: 25
    top_k_modes: 5
    nhead: 8
    dim_feedforward: 512
    weight_decay: 0.001
    scheduler_patience: 5
    scheduler_factor: 0.5
    clip_grad_norm: 1.0
    partially_finite_target: false
  lr: 0.001
  optimizer: AdamW
  lr_policy: constant

[2025-10-05 04:24:24,991][root][INFO] - Set random seed to 42 for reproducibility.
[2025-10-05 04:24:24,991][root][INFO] - CUDA version: 12.1
[2025-10-05 04:24:25,029][root][INFO] - Device count: 1
[2025-10-05 04:24:25,032][root][INFO] - Device name: NVIDIA GeForce RTX 4060 Laptop GPU
[2025-10-05 04:24:25,144][root][INFO] - Loading train.csv from ../data/train.csv
[2025-10-05 04:24:25,223][root][INFO] - Loading train_labels.csv from ../data/train_labels.csv
[2025-10-05 04:24:25,333][root][INFO] - Loading target_pairs.csv from ../data/target_pairs.csv
[2025-10-05 04:24:25,486][root][INFO] - X_full NaNs after imputation: 0
[2025-10-05 04:24:25,486][root][INFO] - Y_full NaNs before imputation: 0
[2025-10-05 04:24:25,488][root][INFO] - Y_full NaNs after imputation: 0
[2025-10-05 04:24:25,488][root][INFO] - X_full feature count: 1671
[2025-10-05 04:24:25,496][root][INFO] - Data loaded: X_full_np shape=(1917, 1671), Y_full_np shape=(1917, 424)
[2025-10-05 04:24:25,496][root][INFO] - Raw data shapes: X=(1917, 1671), Y=(1917, 424)
[2025-10-05 04:24:25,496][root][INFO] - Sample batch shapes: X=(64, 1671), Y=(64, 424)
[2025-10-05 04:24:25,930][root][INFO] - Trial 8 started with params: {'input_size': 1671, 'output_size': 424, 'output_len': 5, 'input_len': 64, 'label_len': 16, 'hidden_size': 128, 'num_layers': 4, 'd_layers': 3, 'nhead': 4, 'dim_feedforward': 1024, 'dropout': 0.25, 'batch_size': 16, 'attn': 'prob', 'distil': True, 'mix': True}, lr=0.00021328622620618546, optimizer=Adam, lr_policy=constant
[2025-10-05 04:24:26,256][root][INFO] - Performed memory cleaning.
[2025-10-05 04:24:26,378][root][INFO] - Trial 9 started with params: {'input_size': 1671, 'output_size': 424, 'output_len': 5, 'input_len': 80, 'label_len': 16, 'hidden_size': 384, 'num_layers': 2, 'd_layers': 3, 'nhead': 8, 'dim_feedforward': 1792, 'dropout': 0.05, 'batch_size': 64, 'attn': 'prob', 'distil': True, 'mix': True}, lr=0.0006899589521075098, optimizer=Adam, lr_policy=constant
[2025-10-05 04:24:26,651][root][INFO] - GPU Memory: 1.06 GB used out of 8.00 GB
[2025-10-05 04:24:26,651][root][INFO] - Performed memory cleaning.
[2025-10-05 04:24:26,651][root][INFO] - GPU Memory: 1.06 GB used out of 8.00 GB
[2025-10-05 04:24:26,655][root][INFO] - Trial 9: X_raw shape=(1917, 1671), Y_raw shape=(1917, 424), NaNs X=0, Y=0
[2025-10-05 04:24:26,655][root][INFO] - Building Informer datasets with input_len=80, output_len=5, label_len=16, use_time_feats=True
[2025-10-05 04:24:26,656][root][INFO] - Trial 8: X_raw shape=(1917, 1671), Y_raw shape=(1917, 424), NaNs X=0, Y=0
[2025-10-05 04:24:26,656][root][INFO] - Building Informer datasets with input_len=64, output_len=5, label_len=16, use_time_feats=True
[2025-10-05 04:24:27,061][root][INFO] - WindowDataset created: Xw shape=(1849, 64, 1671), Yw shape=(1849, 5, 424), invalid_windows: 0
[2025-10-05 04:24:27,313][root][INFO] - WindowDataset created: Xw shape=(1833, 80, 1671), Yw shape=(1833, 5, 424), invalid_windows: 0
[2025-10-05 04:24:30,635][root][INFO] - InformerWindowDataset created: Xw shape=(1849, 64, 1671), x_dec shape=(1849, 21, 424), Yw shape=(1849, 5, 424), mark_enc shape=(1849, 64, 128), invalid_windows=0
[2025-10-05 04:24:30,638][root][INFO] - Informer dataset created: X_enc shape=(1849, 64, 1671), x_dec shape=(1849, 21, 424), Y shape=(1849, 5, 424), mark_enc shape=(1849, 64, 128)
[2025-10-05 04:24:30,858][root][INFO] - Informer DataLoaders created: train=1388, val=277, test=184
[2025-10-05 04:24:30,858][root][INFO] - Trial 8: Creating InTrainer with device=cuda
[2025-10-05 04:24:30,858][root][INFO] - InTrainer initialized with model_kwargs: {'input_size': 1671, 'enc_in': 1671, 'dec_in': 424, 'c_out': 424, 'seq_len': 64, 'label_len': 16, 'out_len': 5, 'd_model': 128, 'n_heads': 4, 'e_layers': 4, 'd_layers': 3, 'd_ff': 1024, 'dropout': 0.25, 'attn': 'prob', 'embed': 'fixed', 'freq': 'd', 'activation': 'gelu', 'distil': True, 'mix': True, 'device': 'cuda'}
[2025-10-05 04:24:30,901][root][INFO] - InformerWindowDataset created: Xw shape=(1833, 80, 1671), x_dec shape=(1833, 21, 424), Yw shape=(1833, 5, 424), mark_enc shape=(1833, 80, 384), invalid_windows=0
[2025-10-05 04:24:30,907][root][INFO] - Informer dataset created: X_enc shape=(1833, 80, 1671), x_dec shape=(1833, 21, 424), Y shape=(1833, 5, 424), mark_enc shape=(1833, 80, 384)
[2025-10-05 04:24:30,930][root][INFO] - Trainer initialized with device=cpu
[2025-10-05 04:24:31,006][root][ERROR] - Trial 8 failed: index 61 is out of bounds for dimension 2 with size 40
Traceback (most recent call last):
  File "/root/vscode/portfolio/mitsui_2025/src/study.py", line 202, in objective_informer
    val_loss = trainer.train(train_loader=train_loader, val_loader=val_loader)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/trainer.py", line 673, in train
    pred = self.model(x_enc, x_mark_enc=x_mark_enc, x_dec=x_dec, x_mark_dec=x_mark_dec)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1561, in forward
    enc_out, attns = self.encoder(enc_out, attn_mask=enc_self_mask)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1379, in forward
    x, attn = attn_layer(x, attn_mask=attn_mask)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1361, in forward
    new_x, attn = self.attention(x, x, x, attn_mask=attn_mask)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1342, in forward
    out, attn = self.inner_attention(queries, keys, values, attn_mask)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1315, in forward
    context, attn = self._update_context(context, values, scores_top, index, u, attn_mask)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1291, in _update_context
    context_in[torch.arange(B)[:, None, None], torch.arange(H)[None, :, None], index, :] = torch.matmul(attn, V).type_as(context_in)
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IndexError: index 61 is out of bounds for dimension 2 with size 40

[2025-10-05 04:24:31,325][root][INFO] - GPU Memory: 1.06 GB used out of 8.00 GB
[2025-10-05 04:24:31,604][root][INFO] - Performed memory cleaning.
[2025-10-05 04:24:31,662][root][INFO] - Informer DataLoaders created: train=1376, val=274, test=183
[2025-10-05 04:24:31,662][root][INFO] - Trial 9: Creating InTrainer with device=cuda
[2025-10-05 04:24:31,663][root][INFO] - InTrainer initialized with model_kwargs: {'input_size': 1671, 'enc_in': 1671, 'dec_in': 424, 'c_out': 424, 'seq_len': 80, 'label_len': 16, 'out_len': 5, 'd_model': 384, 'n_heads': 8, 'e_layers': 2, 'd_layers': 3, 'd_ff': 1792, 'dropout': 0.05, 'attn': 'prob', 'embed': 'fixed', 'freq': 'd', 'activation': 'gelu', 'distil': True, 'mix': True, 'device': 'cuda'}
[2025-10-05 04:24:31,749][root][INFO] - Trainer initialized with device=cpu
[2025-10-05 04:24:31,999][root][ERROR] - Trial 9 failed: index 71 is out of bounds for dimension 2 with size 40
Traceback (most recent call last):
  File "/root/vscode/portfolio/mitsui_2025/src/study.py", line 202, in objective_informer
    val_loss = trainer.train(train_loader=train_loader, val_loader=val_loader)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/trainer.py", line 673, in train
    pred = self.model(x_enc, x_mark_enc=x_mark_enc, x_dec=x_dec, x_mark_dec=x_mark_dec)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1561, in forward
    enc_out, attns = self.encoder(enc_out, attn_mask=enc_self_mask)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1379, in forward
    x, attn = attn_layer(x, attn_mask=attn_mask)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1361, in forward
    new_x, attn = self.attention(x, x, x, attn_mask=attn_mask)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1342, in forward
    out, attn = self.inner_attention(queries, keys, values, attn_mask)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/mitsui/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1315, in forward
    context, attn = self._update_context(context, values, scores_top, index, u, attn_mask)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/vscode/portfolio/mitsui_2025/src/models.py", line 1291, in _update_context
    context_in[torch.arange(B)[:, None, None], torch.arange(H)[None, :, None], index, :] = torch.matmul(attn, V).type_as(context_in)
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IndexError: index 71 is out of bounds for dimension 2 with size 40

[2025-10-05 04:24:32,372][root][INFO] - GPU Memory: 1.06 GB used out of 8.00 GB
[2025-10-05 04:24:32,644][root][INFO] - Performed memory cleaning.
[2025-10-05 04:24:32,665][root][WARNING] - No completed trials for INF. All trials were pruned or failed. Trials: 10
[2025-10-05 04:24:32,667][root][INFO] - Trial states: [3, 3, 3, 3, 2, 2, 3, 3, 2, 2]
